examples/language-modeling/run_clm.py

```diff
@@ -547,15 +547,15 @@ def group_texts(examples):
         if "train" not in tokenized_datasets:
             raise ValueError("--do_train requires a train dataset")
         train_dataset = lm_datasets["train"]
-        if data_args.max_train_samples is not None:
+        if data_args.max_train_samples is not None and not data_args.streaming:
             max_train_samples = min(len(train_dataset), data_args.max_train_samples)
             train_dataset = train_dataset.select(range(max_train_samples))
 
     if training_args.do_eval:
         if "validation" not in tokenized_datasets:
             raise ValueError("--do_eval requires a validation dataset")
         eval_dataset = lm_datasets["validation"]
-        if data_args.max_eval_samples is not None:
+        if data_args.max_eval_samples is not None and not data_args.streaming:
             max_eval_samples = min(len(eval_dataset), data_args.max_eval_samples)
             eval_dataset = eval_dataset.select(range(max_eval_samples))
```

Comment:
`max_eval_samples` will actually not have any effect in streaming mode since this line cannot be computed with an iterable dataset.